auth/src/main/java/com/vimeo/networking2/internal/AuthenticatorImpl.kt

```diff
@@ -230,24 +230,77 @@ internal class AuthenticatorImpl(
 
             call.enqueueError(apiError, callback)
         } else {
-            call.enqueue(callback)
+            call.enqueueWithAccountStore(callback)
         }
     }
 
     override fun authenticateWithCodeGrant(uri: String, callback: VimeoCallback<VimeoAccount>): VimeoRequest {
         val httpUrl: HttpUrl = HttpUrl.parse(uri) ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
-        val authorizationCode = httpUrl.queryParameter("code")
+        val authorizationCode = httpUrl.queryParameter(PARAM_CODE)
             ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
         return authService.authenticateWithCodeGrant(
             basicAuthHeader,
             redirectUri,
             authorizationCode,
             GrantType.AUTHORIZATION_CODE
-        ).enqueue(AccountStoringVimeoCallback(accountStore, callback))
+        ).enqueueWithAccountStore(callback)
     }
 
     override fun createCodeGrantAuthorizationUri(responseCode: String): String {
-        return authService.codeGrantRequest(clientId, redirectUri, responseCode, scopes).request().url().toString()
+        return authService.codeGrantRequest(clientId, redirectUri, responseCode, scopes).url
+    }
+
+    override fun fetchSsoDomain(domain: String, callback: VimeoCallback<SsoDomain>): VimeoRequest {
+        val invalidAuthParams = mapOf(
+            AuthParam.DOMAIN to domain
+        ).validate()
+
+        val call = authService.getSsoDomain(basicAuthHeader, domain)
+
+        return if (invalidAuthParams.isNotEmpty()) {
+            val apiError = ApiError(
+                developerMessage = "SSO domain fetch error",
+                invalidParameters = invalidAuthParams
+            )
+            call.enqueueError(apiError, callback)
+        } else {
+            call.enqueue(callback)
+        }
+    }
+
+    override fun createSsoAuthorizationUri(ssoDomain: SsoDomain, responseCode: String): String {
+        val connectUrl = requireNotNull(ssoDomain.connectUrl) { "SsoDomain.connectUrl cannot be null" }
```

Comment:
This function just creates a URL, it doesn't actually make a request, so there is no callback to notify of an error. Our only option is to throw an exception or return null. I will update the docs on the function so that it reflects that a non null URL is not allowed.